### PR TITLE
Change validators to use k8s:<name> tags, don't generate for core

### PR DESCRIPTION
**Pull request by thockin**
This will _not_ generate for the tags which already exist in k/k, making adoption much more incremental.

### CODE REVIEW

File: staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/all_types_match/trivial/testdata/validate-false.json [hunk 1]
```diff
@@ -1,6 +1 @@
-{
-    "*trivial.E1": {},
-    "*trivial.E2": {},
-    "*trivial.T1": {},
-    "*trivial.T2": {}
-  }
\ No newline at end of file
+{}
```
Comment (aaron-prindle) [inline staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/all_types_match/trivial/testdata/validate-false.json:1]: nit: is this validate-false.json file intended to have an empty struct?

Comment (thockin) [reply staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/all_types_match/trivial/testdata/validate-false.json:1]: This was auto-generated by `UPDATE_VALIDATION_GEN_FIXTURE_DATA=true go test ./staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/...`

Comment (aaron-prindle) [reply staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/all_types_match/trivial/testdata/validate-false.json:1]: Ah got it, thanks

**Review by aaron-prindle — APPROVED** [review]
LGTM, thanks!

File: staging/src/k8s.io/code-generator/cmd/validation-gen/validators/openapi.go [hunk 2]
```diff
@@ -72,8 +72,11 @@ func (openAPIDeclarativeValidator) ExtractValidations(t *types.Type, comments []
 			return result, fmt.Errorf("multiple %s tags found", formatTagName)
 		}
 		format := formats[0]
-		formatFunction := FormatValidationFunction(format)
-		if formatFunction != nil {
+		if formatFunction, err := getFormatValidationFunction(format); err != nil {
+			return result, err
+		} else if formatFunction != nil {
```
Comment (yongruilin) [inline staging/src/k8s.io/code-generator/cmd/validation-gen/validators/openapi.go:73]: Shouldn't it be `formatFunction == nil`?

Comment (thockin) [reply staging/src/k8s.io/code-generator/cmd/validation-gen/validators/openapi.go:73]: Clearly so, dangit :)  Will send a new one

Comment (thockin) [reply staging/src/k8s.io/code-generator/cmd/validation-gen/validators/openapi.go:73]: again, bitten by lack of tests.

**Pull request merged**